### PR TITLE
MVS 502 Adding to PID and other segments

### PR DESCRIPTION
**Pull request by nxm9892**
## :floppy_disk: [Code Check]
_Make sure you've checked/done the following before making this PR!_
* Does your code build/run? Yes
* Does your design approach make sense? Yes
* Does your work achieve the intention of the card? Yes.
* Does your code have any unintentional side effects -- have you tested a full workflow & checked the logic branches you touched? Yes
* Did you make a unit test for your code (as appropriate)? N/A
* Is your branch relatively up to date with the Development branch (within ~2 commits) Yes
* Have you moved the card to "Needs QA" in Jira About to

## :flipper: [JIRA Sprint Card]

https://mass-immunization-system.atlassian.net/browse/MVS-502
https://mass-immunization-system.atlassian.net/browse/MVS-503
https://mass-immunization-system.atlassian.net/browse/MVS-530

## :newspaper: [Work Description]

1. Added additional code to the PID mapping in the Mirth channel
2. Added mappings for the other segments 
3. Commented out some of the other mappings due to some errors

Update:
1. Added additional code for ORC, RXA, RXR segments.
2. Updated PID segment according to requested Q/A changes.
3. Added additional resources to fhir bundle sample message.
4. Uncommented code on RXA segment 

## :vertical_traffic_light: [Testing/QA Notes]

1. First, install Mirth on your computer and get it running: http://www.mirthcorp.com/community/wiki/display/mirth/Getting+Started+Guide. Let me know if you need help with this step.

2. Import the channel added by this PR, enable it, and deploy it.

3. Copy the template JSON, then right click on the channel in the dashboard, click Send Message, then paste the JSON into the window that comes up.

4. Right click the channel again, go to View Messages, and confirm that the message that you sent processed without errors and filled in the PID, ORC, RXA, RXR WIR-required fields.

### CODE REVIEW

File: NextGenConnect/JSON to HL7v2.xml [hunk 1]
```diff
@@ -83,35 +83,257 @@ for (var i = 0; i &lt; getArrayOrXmlLength(patient[&apos;identifier&apos;]); i++
...
-/*
 //PID-5 patient name
-tmp[&apos;PID&apos;][&apos;PID.5&apos;][&apos;PID.5.1&apos;] = patient[&apos;name&apos;][0][&apos;family&apos;][&apos;@value&apos;].toString();
-tmp[&apos;PID&apos;][&apos;PID.5&apos;][&apos;PID.5.2&apos;] = patient[&apos;name&apos;][0][&apos;given&apos;][&apos;@value&apos;].toString();
-tmp[&apos;PID&apos;][&apos;PID.5&apos;][&apos;PID.5.3&apos;] = &quot;&quot;; //not in XML
+tmp[&apos;PID&apos;][&apos;PID.5&apos;][&apos;PID.5.1&apos;] = patient[&apos;name&apos;][0][&apos;family&apos;];
+tmp[&apos;PID&apos;][&apos;PID.5&apos;][&apos;PID.5.2&apos;] = patient[&apos;name&apos;][0][&apos;given&apos;][0]; //Why is this stored in an array in sample JSON message?
+tmp[&apos;PID&apos;][&apos;PID.5&apos;][&apos;PID.5.7&apos;] = &quot;L&quot;;  // Legal Name		
+
+//PID-6 Mothers Maiden Name -- ??? Try to grab value ?
```
Comment (rrucker3) [inline NextGenConnect/JSON to HL7v2.xml:91]: If we aren't collecting this in the registration app, we can leave this out. Check with front end.

File: NextGenConnect/JSON to HL7v2.xml [hunk 2]
```diff
@@ -83,35 +83,257 @@ for (var i = 0; i &lt; getArrayOrXmlLength(patient[&apos;identifier&apos;]); i++
...
-tmp[&apos;PID&apos;][&apos;PID.7&apos;][&apos;PID.7.1&apos;] = patient[&apos;birthDate&apos;][&apos;@value&apos;].toString();
-tmp[&apos;PID&apos;][&apos;PID.8&apos;][&apos;PID.8.1&apos;] = patient[&apos;gender&apos;][&apos;@value&apos;].toString();
 
 //PID-11 address
-tmp[&apos;PID&apos;][&apos;PID.11&apos;][&apos;PID.11.1&apos;] = patient[&apos;address&apos;][&apos;line&apos;][&apos;@value&apos;].toString();
-tmp[&apos;PID&apos;][&apos;PID.11&apos;][&apos;PID.11.7&apos;] = patient[&apos;address&apos;][&apos;use&apos;][&apos;@value&apos;].toString();
-tmp[&apos;PID&apos;][&apos;PID.11&apos;][&apos;PID.11.3&apos;] = patient[&apos;address&apos;][&apos;city&apos;][&apos;@value&apos;].toString();
-tmp[&apos;PID&apos;][&apos;PID.11&apos;][&apos;PID.11.4&apos;] = patient[&apos;address&apos;][&apos;state&apos;][&apos;@value&apos;].toString();
-tmp[&apos;PID&apos;][&apos;PID.11&apos;][&apos;PID.11.5&apos;] = patient[&apos;address&apos;][&apos;postalCode&apos;][&apos;@value&apos;].toString();
+tmp[&apos;PID&apos;][&apos;PID.11&apos;][&apos;PID.11.1&apos;] = patient[&apos;address&apos;][0][&apos;line&apos;][0];
```
Comment (rrucker3) [inline NextGenConnect/JSON to HL7v2.xml:120]: Taking only the first line might leave an incomplete address, like if the patient lives in an apartment or something. Can you do a .join with a comma like in PID-3 so we get all the lines?

File: NextGenConnect/JSON to HL7v2.xml [hunk 3]
```diff
@@ -83,35 +83,257 @@ for (var i = 0; i &lt; getArrayOrXmlLength(patient[&apos;identifier&apos;]); i++
...
+tmp[&apos;PID&apos;][&apos;PID.11&apos;][&apos;PID.11.3&apos;] = patient[&apos;address&apos;][0][&apos;city&apos;];
+tmp[&apos;PID&apos;][&apos;PID.11&apos;][&apos;PID.11.4&apos;] = patient[&apos;address&apos;][0][&apos;state&apos;];
+tmp[&apos;PID&apos;][&apos;PID.11&apos;][&apos;PID.11.5&apos;] = patient[&apos;address&apos;][0][&apos;postalCode&apos;];
+tmp[&apos;PID&apos;][&apos;PID.11&apos;][&apos;PID.11.6&apos;] = patient[&apos;address&apos;][0][&apos;country&apos;];
+tmp[&apos;PID&apos;][&apos;PID.11&apos;][&apos;PID.11.7&apos;] = &apos;M&apos; // Mailing address
 
-//PID-13 Home Phone Number
-tmp[&apos;PID&apos;][&apos;PID.13&apos;][&apos;PID.13.1&apos;] = patient[&apos;telecom&apos;][0][&apos;value&apos;][&apos;@value&apos;].toString();
+//PID-13 Home Phone Number -- Need to seperate the area code and number if sending...
+tmp[&apos;PID&apos;][&apos;PID.13&apos;][&apos;PID.13.1&apos;] = patient[&apos;telecom&apos;][0][&apos;value&apos;];
```
Comment (rrucker3) [inline NextGenConnect/JSON to HL7v2.xml:128]: You'll need to parse the area code out or WIR won't take it. Try something like this: https://stackoverflow.com/questions/13868130/parse-area-code-out-of-phone-number

File: NextGenConnect/JSON to HL7v2.xml [hunk 4]
```diff
@@ -83,35 +83,257 @@ for (var i = 0; i &lt; getArrayOrXmlLength(patient[&apos;identifier&apos;]); i++
...
+
+// NK1-1 Set ID - Sets the ID for the first related party and so on 1..2..3 for multiple NK1 segments
+// If only listing one contact this always remains 1
+
+tmp[&apos;NK1&apos;][&apos;NK1.1&apos;] = 1;
+
+//NK1-2 Name of responsible person
+
+tmp[&apos;NK1&apos;][&apos;NK1.2&apos;][&apos;NK1.2.1&apos;] = patient[&apos;contact&apos;][1][&apos;family&apos;];
+tmp[&apos;NK1&apos;][&apos;NK1.2&apos;][&apos;NK1.2.2&apos;] = patient[&apos;contact&apos;][1][&apos;given&apos;][0]; // Stored in array??
```
Comment (rrucker3) [inline NextGenConnect/JSON to HL7v2.xml:183]: Same as PID: it's for middle names.

File: NextGenConnect/JSON to HL7v2.xml [hunk 5]
```diff
@@ -83,35 +83,257 @@ for (var i = 0; i &lt; getArrayOrXmlLength(patient[&apos;identifier&apos;]); i++
...
+// if relationship == cargeiver, this = CGV, and so forth
+
+
+tmp[&apos;NK1&apos;][&apos;NK1.3&apos;][&apos;NK1.3.2&apos;] = &apos;Mother&apos;; // Changes depending on code ^^^^
+
+tmp[&apos;NK1&apos;][&apos;NK1.3&apos;][&apos;NK1.3.3&apos;] = &apos;HL70063&apos;; // This is the default table... maybe find one compatible with FHIR 0131
+
+//NK1-4 Address of responsible person
+
+tmp[&apos;NK1&apos;][&apos;NK1.4&apos;][&apos;NK1.4.2&apos;] = patient[&apos;contact&apos;][3][&apos;line&apos;][0];
```
Comment (rrucker3) [inline NextGenConnect/JSON to HL7v2.xml:201]: Same as PID: we need all the lines, else it might not be right.

File: NextGenConnect/JSON to HL7v2.xml [hunk 6]
```diff
@@ -83,35 +83,257 @@ for (var i = 0; i &lt; getArrayOrXmlLength(patient[&apos;identifier&apos;]); i++
...
+tmp[&apos;NK1&apos;][&apos;NK1.4&apos;][&apos;NK1.4.3&apos;] = patient[&apos;contact&apos;][3][&apos;city&apos;];
+tmp[&apos;NK1&apos;][&apos;NK1.4&apos;][&apos;NK1.4.4&apos;] = patient[&apos;contact&apos;][3][&apos;state&apos;];
+tmp[&apos;NK1&apos;][&apos;NK1.4&apos;][&apos;NK1.4.5&apos;] = patient[&apos;contact&apos;][3][&apos;postalCode&apos;];
+tmp[&apos;NK1&apos;][&apos;NK1.4&apos;][&apos;NK1.4.6&apos;] = patient[&apos;contact&apos;][3][&apos;country&apos;];
+tmp[&apos;NK1&apos;][&apos;NK1.4&apos;][&apos;NK1.4.7&apos;]= &apos;M&apos; // Mailing address
+
+//NK1-5 Phone number -- need to seperate area code and number
+
+tmp[&apos;NK1&apos;][&apos;NK1.5&apos;][&apos;NK1.5.2&apos;] = patient[&apos;contact&apos;][2][&apos;value&apos;];
```
Comment (rrucker3) [inline NextGenConnect/JSON to HL7v2.xml:210]: Same as PID, parse it out. Maybe the parsing code could go in a code template or a global script? I haven't explored those too much. Maybe that's how we share code across transformers instead of copy/pasting.

File: NextGenConnect/JSON to HL7v2.xml [hunk 7]
```diff
@@ -83,35 +83,257 @@ for (var i = 0; i &lt; getArrayOrXmlLength(patient[&apos;identifier&apos;]); i++
...
+tmp[&apos;PID&apos;][&apos;PID.5&apos;][&apos;PID.5.2&apos;] = patient[&apos;name&apos;][0][&apos;given&apos;][0]; //Why is this stored in an array in sample JSON message?
+tmp[&apos;PID&apos;][&apos;PID.5&apos;][&apos;PID.5.7&apos;] = &quot;L&quot;;  // Legal Name		
+
+//PID-6 Mothers Maiden Name -- ??? Try to grab value ?
+
+//PID-7 Birthdate 
+
+var birthDate = patient[&apos;birthDate&apos;];
+
+birthDate.replace(&apos;-&apos;,&apos;&apos;); // Replaces -&apos;s in birthDate
```
Comment (rrucker3) [inline NextGenConnect/JSON to HL7v2.xml:97]: Also, I'm not sure why, but dashes were included in the HL7 message when they shouldn't be. Maybe .replace doesn't work in-place in javascript.

**Review by rrucker3 — CHANGES_REQUESTED** [review]
Back to you. I left some comments. Let me know if you want to chat about my feedback.

File: NextGenConnect/JSON to HL7v2.xml [hunk 8]
```diff
@@ -83,35 +83,257 @@ for (var i = 0; i &lt; getArrayOrXmlLength(patient[&apos;identifier&apos;]); i++
...
+
+//ORC-1 Order Control - Must contatin value &apos;RE&apos;
+tmp[&apos;ORC&apos;][&apos;ORC.1&apos;] = &apos;RE&apos;;
+
+//ORC-2 Placer Order Number (RE) -- Not required
+
+
+//ORC-3 Filler Order Number(R) -- Is this the lot number in immunization resource? Where can I find this?
+
+tmp[&apos;ORC&apos;][&apos;ORC.3&apos;] = msg[&apos;entry&apos;][0][&apos;resource&apos;][&apos;id&apos;];
```
[the remark this answers is not in the document]
Comment (nxm9892) [reply NextGenConnect/JSON to HL7v2.xml:231]: Good point

File: NextGenConnect/JSON to HL7v2.xml [hunk 9]
```diff
@@ -83,35 +83,257 @@ for (var i = 0; i &lt; getArrayOrXmlLength(patient[&apos;identifier&apos;]); i++
...
+    return entry[&apos;resource&apos;][&apos;resourceType&apos;] === &quot;Immunization&quot;;
+})[0][&apos;resource&apos;];
+
+//ORC-1 Order Control - Must contatin value &apos;RE&apos;
+tmp[&apos;ORC&apos;][&apos;ORC.1&apos;] = &apos;RE&apos;;
+
+//ORC-2 Placer Order Number (RE) -- Not required
+
+
+//ORC-3 Filler Order Number(R) -- Is this the lot number in immunization resource? Where can I find this?
```
[the remark this answers is not in the document]
Comment (nxm9892) [reply NextGenConnect/JSON to HL7v2.xml:229]: Nope. Deleted.

File: NextGenConnect/JSON to HL7v2.xml (hunk 7 above)
[the remark this answers is not in the document]
Comment (nxm9892) [reply NextGenConnect/JSON to HL7v2.xml:97]: Looking into that.

File: NextGenConnect/JSON to HL7v2.xml [hunk 10]
```diff
@@ -83,35 +83,257 @@ for (var i = 0; i &lt; getArrayOrXmlLength(patient[&apos;identifier&apos;]); i++
...
 tmp[&apos;PID&apos;][&apos;PID.3&apos;] = ids.join(&apos;~&apos;);
 
-
-/*
 //PID-5 patient name
-tmp[&apos;PID&apos;][&apos;PID.5&apos;][&apos;PID.5.1&apos;] = patient[&apos;name&apos;][0][&apos;family&apos;][&apos;@value&apos;].toString();
-tmp[&apos;PID&apos;][&apos;PID.5&apos;][&apos;PID.5.2&apos;] = patient[&apos;name&apos;][0][&apos;given&apos;][&apos;@value&apos;].toString();
-tmp[&apos;PID&apos;][&apos;PID.5&apos;][&apos;PID.5.3&apos;] = &quot;&quot;; //not in XML
+tmp[&apos;PID&apos;][&apos;PID.5&apos;][&apos;PID.5.1&apos;] = patient[&apos;name&apos;][0][&apos;family&apos;];
+tmp[&apos;PID&apos;][&apos;PID.5&apos;][&apos;PID.5.2&apos;] = patient[&apos;name&apos;][0][&apos;given&apos;][0]; //Why is this stored in an array in sample JSON message?
```
[the remark this answers is not in the document]
Comment (nxm9892) [reply NextGenConnect/JSON to HL7v2.xml:88]: Forgot to delete that comment. Noted.

File: NextGenConnect/JSON to HL7v2.xml (hunk 1 above)
Comment (nxm9892) [reply NextGenConnect/JSON to HL7v2.xml:91]: Deleted.

File: NextGenConnect/JSON to HL7v2.xml (hunk 3 above)
Comment (nxm9892) [reply NextGenConnect/JSON to HL7v2.xml:128]: Thanks for the link. Will sort this out.

File: NextGenConnect/JSON to HL7v2.xml (hunk 7 above)
Comment (nxm9892) [reply NextGenConnect/JSON to HL7v2.xml:97]: It probably doesn't I'm going to look for a different function

File: NextGenConnect/JSON to HL7v2.xml [hunk 11]
```diff
@@ -253,21 +244,40 @@ var location = msg[&apos;entry&apos;].filter(function (entry){
...
 
-//RXA-3 Data of Administration -- Need to worry if time component stored 
-//tmp[&apos;RXA&apos;][&apos;RXA.3&apos;] = immunization[&apos;occurenceDateTime&apos;].replace(&apos;-&apos;,&apos;&apos;); // commented out due to undefined
+//RXA-3 Data of Administrationmsg[&apos;entry&apos;][0][&apos;resource&apos;][&apos;occurrenceDateTime&apos;]
+
+var occurenceDate = immunization[&apos;occurrenceDateTime&apos;];
+tmp[&apos;RXA&apos;][&apos;RXA.3&apos;] = occurenceDate.split(&apos;-&apos;).join(&apos;&apos;); 
 
-//RXA-5 Administered Code -- Needs to be CVX -- how are we converting what&apos;s stored in FHIR resource? Hardcode?
+//RXA-5 Administered Code -- // NDC code?
```
Comment (rrucker3) [inline NextGenConnect/JSON to HL7v2.xml:268]: Are we handling RXA-5 in this card? If not, is there a card in the backlog to deal with it?

File: NextGenConnect/JSON to HL7v2.xml [hunk 12]
```diff
@@ -253,21 +244,40 @@ var location = msg[&apos;entry&apos;].filter(function (entry){
...
 
-// If/switch  statements hard code this in... cvx codes for coronavirus ....
-tmp[&apos;RXA&apos;][&apos;RXA.5.1&apos;][&apos;RXA.5.1.1&apos;]  = immunization[&apos;vaccineCode&apos;][&apos;coding&apos;][0][&apos;code&apos;];
-tmp[&apos;RXA&apos;][&apos;RXA.5.1&apos;][&apos;RXA.5.1.2&apos;]  = immunization[&apos;vaccineCode&apos;][&apos;coding&apos;][0][&apos;display&apos;];
+/* Need to add organization to bundle
```
Comment (rrucker3) [inline NextGenConnect/JSON to HL7v2.xml:270]: Any reason not to add these as part of this PR? Or are you leaving it for later? If so, is there a card in the backlog?

File: NextGenConnect/JSON to HL7v2.xml [hunk 13]
```diff
@@ -277,19 +287,17 @@ tmp[&apos;RXA&apos;][&apos;RXA.6&apos;] = immunization[&apos;doseQuantity&apos;]
...
 tmp[&apos;RXA&apos;][&apos;RXA.7&apos;] = immunization[&apos;doseQuantity&apos;][&apos;code&apos;];
 
-//RXA-9 Administration Notes 00 -&gt; Administered, 01 -&gt; Record
+//RXA-9 Administration Notes 
 
 tmp[&apos;RXA&apos;][&apos;RXA.9&apos;] = 00;
 
 
-//RXA-10 Administering Provider -- Need a reference to actor
+//RXA-10 Administering Provider -- Reference to Practioner
```
Comment (rrucker3) [inline NextGenConnect/JSON to HL7v2.xml:295]: Similarly to Organization above, is this in scope for this PR?

**Review by rrucker3 — CHANGES_REQUESTED** [review]
In addition to these comments, I noticed a few comments from my first review (about addresses) were not resolved. Could you take a look at those comments? Also, for future reference, just reply to a comment; don't resolve it. In my experience, the QAer that made the comment resolves it once it has been addressed to the satisfaction of the reviewer.

You need to either rebase this branch on Development or merge in the changes using the Resolve conflicts button in Github, since I made some changes to NextGenConnect/JSON to HL7v2.xml in a branch I just merged in.

If you are going to add MVS-503 to this PR (which is fine), could you add information about it to the PR info box? Testing information, link to the card, etc.

If you want to have a discussion about any of this, please let me know.

File: NextGenConnect/JSON to HL7v2.xml (hunk 11 above)
Comment (nxm9892) [reply NextGenConnect/JSON to HL7v2.xml:268]: It is handled. WIR specification suggests we include the NDC code too (5.1.4-5.1.6) although it is not necessary. However the 5.1.1-5.1.3 have inputs.

File: NextGenConnect/JSON to HL7v2.xml (hunk 12 above)
Comment (nxm9892) [reply NextGenConnect/JSON to HL7v2.xml:270]: There is a card to add the extra resources to the bundle in the backlog.

File: NextGenConnect/JSON to HL7v2.xml (hunk 13 above)
Comment (nxm9892) [reply NextGenConnect/JSON to HL7v2.xml:295]: There is a card to add the extra resources to the bundle in the backlog.

File: NextGenConnect/JSON to HL7v2.xml (hunk 7 above)
[the remark this answers is not in the document]
Comment (nxm9892) [reply NextGenConnect/JSON to HL7v2.xml]: This was fixed.

File: NextGenConnect/JSON to HL7v2.xml (hunk 8 above)
[the remark this answers is not in the document]
Comment (nxm9892) [reply NextGenConnect/JSON to HL7v2.xml]: This was also fixed.

File: NextGenConnect/JSON to HL7v2.xml (hunk 2 above)
Comment (nxm9892) [reply NextGenConnect/JSON to HL7v2.xml:115]: I don't really see the point of joining these. Taking the first line would signify the primary address, would it not? The second line being a secondary... etc. I'd assume a patient with an apartment address would be the same.

File: NextGenConnect/JSON to HL7v2.xml (hunk 3 above)
Comment (nxm9892) [reply NextGenConnect/JSON to HL7v2.xml]: Done.

File: NextGenConnect/JSON to HL7v2.xml (hunk 7 above)
Comment (nxm9892) [reply NextGenConnect/JSON to HL7v2.xml]: This is done.

File: NextGenConnect/JSON to HL7v2.xml (hunk 2 above)
Comment (nxm9892) [reply NextGenConnect/JSON to HL7v2.xml:115]: At least as far as I have seen in every sample fhir patient resource I've looked at.

**Comment by nxm9892** [comment]
I unresolved most of the comments, and updated them. For MVS-503 I can add info about it in the the PR. I believe the only discussion that really needs to be had is on the address comment. As I am under a different assumption.

File: NextGenConnect/JSON to HL7v2.xml (hunk 5 above)
Comment (nxm9892) [reply NextGenConnect/JSON to HL7v2.xml:192]: I just left this as we are not worrying about NK1

File: NextGenConnect/JSON to HL7v2.xml (hunk 6 above)
Comment (nxm9892) [reply NextGenConnect/JSON to HL7v2.xml:201]: I just left this as we are not worrying about NK1

File: NextGenConnect/JSON to HL7v2.xml (hunk 2 above)
Comment (rrucker3) [reply NextGenConnect/JSON to HL7v2.xml:115]: I think you're confusing patient['address'], which is an array of address objects with patient['address'][n]['line'], which is an array of strings. Each of the address objects in patient['address'] can have an array of lines that looks like this:

Line 1: 1234 Main Street
Line 2: Apartment 502

If we just take the first address line, which is what you're doing if you only get patient['address'][0]['line'][0], you won't get the apartment number, suite, or whatever else people put on the subsequent address lines.

File: NextGenConnect/JSON to HL7v2.xml (hunk 5 above)
Comment (rrucker3) [reply NextGenConnect/JSON to HL7v2.xml:192]: Might as well fix it while we're here, right? You can just use the same solution as the one for PID, so it won't be much extra effort.

File: NextGenConnect/JSON to HL7v2.xml (hunk 11 above)
Comment (rrucker3) [reply NextGenConnect/JSON to HL7v2.xml:268]: Where? I don't see it handled in any code that isn't commented out.

File: NextGenConnect/JSON to HL7v2.xml (hunk 12 above)
Comment (rrucker3) [reply NextGenConnect/JSON to HL7v2.xml:270]: Which card is it? Could you include a link to it in your reply?

File: NextGenConnect/JSON to HL7v2.xml (hunk 13 above)
Comment (rrucker3) [reply NextGenConnect/JSON to HL7v2.xml:295]: Which card is it? Could you include a link to it in your reply?

File: NextGenConnect/JSON to HL7v2.xml (hunk 12 above)
Comment (nxm9892) [reply NextGenConnect/JSON to HL7v2.xml:270]: https://mass-immunization-system.atlassian.net/jira/software/projects/MVS/boards/2?selectedIssue=MVS-530

File: NextGenConnect/JSON to HL7v2.xml (hunk 13 above)
Comment (nxm9892) [reply NextGenConnect/JSON to HL7v2.xml:295]: https://mass-immunization-system.atlassian.net/jira/software/projects/MVS/boards/2?selectedIssue=MVS-530

File: NextGenConnect/JSON to HL7v2.xml (hunk 11 above)
Comment (nxm9892) [reply NextGenConnect/JSON to HL7v2.xml:268]: The code is commented out because I need to add the references to the other resources to access the value or else it's undefined.

Comment (nxm9892) [reply NextGenConnect/JSON to HL7v2.xml:268]: See: *Need to add organization to bundle*

File: NextGenConnect/JSON to HL7v2.xml (hunk 2 above)
Comment (nxm9892) [reply NextGenConnect/JSON to HL7v2.xml:115]: I didn't do this yet however I wanted to show you this:
"address": [ {
                "extension": [ {
                    "url": "http://hl7.org/fhir/StructureDefinition/geolocation",
                    "extension": [ {
                        "url": "latitude",
                        "valueDecimal": 42.08145721749866
                    }, {
                        "url": "longitude",
                        "valueDecimal": -87.69894004417375
                    } ]
                } ],
                "line": [ "658 Lueilwitz Spur Apt 55" ],
                "city": "Chicago",
                "state": "IL",
                "postalCode": "60007",
                "country": "US"
            } ]
Does this change your thinking at all?

Comment (nxm9892) [reply NextGenConnect/JSON to HL7v2.xml:115]: It really depends on whether we store it in different elements of the array, which wouldn't make any sense to me. Pretty much all the fhir examples list the "line" out like this.

File: NextGenConnect/JSON to HL7v2.xml (hunk 11 above)
Comment (nxm9892) [reply NextGenConnect/JSON to HL7v2.xml:267]: This is fixed. However need to discuss the NDC code being added.

File: NextGenConnect/JSON to HL7v2.xml (hunk 12 above)
Comment (nxm9892) [reply NextGenConnect/JSON to HL7v2.xml]: This is fixed.

File: NextGenConnect/JSON to HL7v2.xml (hunk 13 above)
Comment (nxm9892) [reply NextGenConnect/JSON to HL7v2.xml]: This is fixed.

**Comment by nxm9892** [comment]
https://mass-immunization-system.atlassian.net/browse/MVS-530 Added the additional resources and uncommented the RXA segment code that was blocked because of it.

File: NextGenConnect/JSON to HL7v2.xml (hunk 2 above)
Comment (rrucker3) [reply NextGenConnect/JSON to HL7v2.xml:115]: This is what it would look like if the patient had multiple addresses, each with multiple address lines (the [] signifies an array in JSON):
"address": [
    {
      "use": "home",
      "line": [
        "Van Egmondkade 23",
        "Apartment 23"
      ],
      "city": "Amsterdam",
      "postalCode": "1024 RJ",
      "country": "NLD"
    },
    {
      "use": "work",
      "line": [
        "Van Egmondkade 24",
        "Suite 101"
      ],
      "city": "Amsterdam",
      "postalCode": "1024 RJ",
      "country": "NLD"
    }
  ]

File: NextGenConnect/JSON to HL7v2.xml [hunk 14]
```diff
@@ -116,34 +106,320 @@ for (var i = 0; i &lt; getArrayOrXmlLength(patient[&apos;identifier&apos;]); i++
...
+	var line = [];
+	line[i] = patient[&apos;address&apos;][0][&apos;line&apos;][i];
+	lines.push(line.join(&apos;, &apos;));
+}
+
 
-//PID-14 Work Phone Number
-tmp[&apos;PID&apos;][&apos;PID.14&apos;][&apos;PID.14.1&apos;] = patient[&apos;telecom&apos;][1][&apos;value&apos;][&apos;@value&apos;].toString();
-*/</script>
+tmp[&apos;PID&apos;][&apos;PID.11&apos;][&apos;PID.11.1&apos;] = lines.join(&apos;&apos;);
```
Comment (rrucker3) [inline NextGenConnect/JSON to HL7v2.xml]: Can this just be like this? It would be simpler than having a for loop. If this won't work, let me know.

`tmp['PID']['PID.11']['PID.11.1'] = patient['address'][0]['line'].join(', ');`

**Review by rrucker3 — APPROVED** [review]
